mod-clans/src/main/java/gg/solarmc/loader/clans/ClanManager.java

```diff
@@ -56,45 +53,59 @@
         return Optional.ofNullable(clans.getIfPresent(id));
     }
 
+
     /**
-     * Gets a clan from memory cache or from table if not present.
-     * @param transaction the tx
-     * @param id ID of clan requested
-     * @return the clan you asked for idiot
-     * @throws IllegalStateException if the clan isn't present in table
+     * Gets a clan by it's identifier
+     *
+     * This fetches from the cache.
+     * @param transaction the transaction
+     * @param id the clan's id
+     * @return the clan if present, empty if not.
      */
-    public Clan getClan(Transaction transaction, int id) {
-        return clans.get(id, i -> {
+    public Optional<Clan> getClanByID(Transaction transaction, int id) {
+        return getClan(transaction, transaction.getProperty(DSLContext.class).fetchOne(CLANS_CLAN_INFO,CLANS_CLAN_INFO.CLAN_ID.eq(id)));
+    }
 
-            var jooq = transaction.getProperty(DSLContext.class);
+    /**
+     * Gets a clan by the name of the clan.
+     *
+     * IMPORTANT: This does not fetch from the cache.
+     * @param transaction the transaction
+     * @param name name of the clan
+     * @return an empty optional if the clan is not present, a clan if it is.
+     */
+    public Optional<Clan> getClanByName(Transaction transaction, String name) {
 
-            ClansClanInfoRecord rec = jooq.fetchOne(CLANS_CLAN_INFO,CLANS_CLAN_INFO.CLAN_ID.eq(i));
+        return getClan(transaction,transaction.getProperty(DSLContext.class).fetchOne(CLANS_CLAN_INFO,CLANS_CLAN_INFO.CLAN_NAME.eq(name)));
+    }
 
-            if (rec == null) {
-                throw new IllegalStateException("No such clan exists!");
-            }
+    //This is very messy. TODO replace this with a server-sided func whenever
+    Optional<Clan> getClan(Transaction transaction, ClansClanInfoRecord record) {
+        var jooq = transaction.getProperty(DSLContext.class);
 
-            Set<ClanMember> bruh = jooq
-                    .select(CLANS_CLAN_MEMBERSHIP.USER_ID).from(CLANS_CLAN_MEMBERSHIP)
-                    .where(CLANS_CLAN_MEMBERSHIP.CLAN_ID.eq(id)).fetchSet((rec1) -> new ClanMember(rec1.value1()));
+        if (record == null) {
+            return Optional.empty();
+        }
 
-            ConcurrentHashMap.KeySetView<ClanMember,Boolean> view = ConcurrentHashMap.newKeySet();
-            view.addAll(bruh);
+        Set<ClanMember> bruh = jooq
```

Comment:
Why do you keep naming local variables `bruh`?

Reply:
sometimes you can't think about variable naming

Reply:
please don't

Reply:
ok i will try to refrain from naming local variables bruh